### PR TITLE
ISPN-6589 ProtobufIndexedFieldProvider may mistakenly report nested fields as unindexed

### PR DESCRIPTION
**Pull request by anistor**
https://issues.jboss.org/browse/ISPN-6589

### CODE REVIEW

File: remote-query/remote-query-server/src/test/java/org/infinispan/query/remote/impl/ProtobufIndexedFieldProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import org.infinispan.test.SingleCacheManagerTest;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.infinispan.transaction.TransactionMode;
+import org.testng.annotations.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author anistor@redhat.com
```
Comment (slaskawi) [inline remote-query/remote-query-server/src/test/java/org/infinispan/query/remote/impl/ProtobufIndexedFieldProviderTest.java]: This javadoc seems redundant - other tests don't have it...

**Comment by slaskawi** [comment]
Only one small comment - otherwise looks ok to me.

**Comment by anistor** [comment]
done.

**Comment by anistor** [comment]
Rebased.

File: remote-query/remote-query-server/src/test/java/org/infinispan/query/remote/impl/ProtobufIndexedFieldProviderTest.java (hunk 1 above)
Comment (danberindei) [inline remote-query/remote-query-server/src/test/java/org/infinispan/query/remote/impl/ProtobufIndexedFieldProviderTest.java]: @slaskawi We need more test comments, not less :)
@anistor How about a short description?

File: remote-query/remote-query-server/src/test/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerInterceptorTest.java [hunk 2]
```diff
@@ -24,7 +24,7 @@
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
-@Test(groups = "functional", testName = "query.remote.ProtobufMetadataManagerInterceptorTest")
+@Test(groups = "functional", testName = "query.remote.impl.ProtobufMetadataManagerInterceptorTest")
```
Comment (danberindei) [inline remote-query/remote-query-server/src/test/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerInterceptorTest.java]: TestNG test names don't need to match the package name exactly, I'd rather leave out the `impl` part.

Comment (anistor) [inline remote-query/remote-query-server/src/test/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerInterceptorTest.java]: I prefer an exact match (minus the org.infinispan prefix),

**Comment by anistor** [comment]
Updated.

**Comment by danberindei** [comment]
Integrated, thanks Adrian!

**Pull request closed without merging**